### PR TITLE
Add filter.js@1.5.2 with git auto-update

### LINKED ISSUE #10391: [Request] Add filter.js
**Library name Filter.js:** 
**Git repository url: https://github.com/jiren/filter.js**

**License(s): MIT license**
**Official homepage:http://jiren.github.io/filter.js/**

=====================
Notes from cdnjs maintainer:
Please read the README.md and CONTRIBUTING.md document first.

You are welcome to add a library via sending pull request,
it'll be faster than just opening a request issue,
and please don't forget to read the guidelines for contributing, thanks!!

### PR DESCRIPTION
**Pull request by clairetsai818**
close #10391, cc @jiren
Pull request for issue: #10391 

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/jiren/filter.js/
 * Official website (optional, not the repository): http://jiren.github.io/filter.js/
 * NPM package url (optional):
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 51
   - Count of stars: 511
   - Count of forks: 512
 * NPM download stats (optional):
   - Downloads in the last day:
   - Downloads in the last week:
   - Downloads in the last month:

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [x] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by clairetsai818** [comment]
@PeterDaveHello Please help me review this, thanks!

File: ajax/libs/filter.js/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    "url": "git://github.com/jiren/filter.js"
+  },
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/jiren/filter.js.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "**/filter.js"
```
Comment (PeterDaveHello) [inline ajax/libs/filter.js/package.json]: We don't use `**/*` under non-dist base path.

**Comment by clairetsai818** [comment]
I removed `**/`. @PeterDaveHello  Please help me review this, thanks!

**Comment by PeterDaveHello** [comment]
@clairetsai818 you should correct the repo url in the comment.

File: ajax/libs/filter.js/1.5.2/filter.min.js [hunk 2]
```diff
@@ -0,0 +1,1727 @@
...
+ * JsonQuery
+ * 0.0.2 (2015-08-06)
+ *
+ * Released under the MIT license
+ * http://opensource.org/licenses/MIT
+ *
+ * Copyright 2011-2015 Jiren Patel[jirenpatel@gmail.com]
+ *
+ */
+
```
Comment (PeterDaveHello) [inline ajax/libs/filter.js/1.5.2/filter.min.js]: Not minified?

File: ajax/libs/filter.js/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    "url": "git://github.com/jiren/filter.js"
+  },
+  "autoupdate": {
+    "source": "git",
+    "target": "git://github.com/jiren/filter.js.git",
+    "fileMap": [
+      {
+        "basePath": "",
+        "files": [
+          "filter.js"
```
Comment (PeterDaveHello) [inline ajax/libs/filter.js/package.json]: Try a more flexible pattern `filter*.+(js|css|map)`

**Comment by clairetsai818** [comment]
@PeterDaveHello I corrected the repo url in the comment, minified the src file and used a flexible pattern. Please help me review this, thanks!

File: ajax/libs/filter.js/1.5.2/filter.js [hunk 4]
```diff
@@ -0,0 +1,1727 @@
+/*
+ * filter.js
+ * 2.1.0 (2016-08-09)
```
Comment (PeterDaveHello) [inline ajax/libs/filter.js/1.5.2/filter.js]: Why `2.1.0` here? Are you sure about what you're doing?

https://github.com/jiren/filter.js/blob/v1.5.2/filter.js#L3

**Comment by clairetsai818** [comment]
@PeterDaveHello I got the latest src file. Please help me review this, thanks!

**Comment by PeterDaveHello** [comment]
@clairetsai818 only src? What about the minified file and map file?

**Comment by clairetsai818** [comment]
@PeterDaveHello I got all of them with the latest version. Please help me review this, thanks!

**Comment by PeterDaveHello** [comment]
@clairetsai818 You don't need to repeat the second sentense ...

**Pull request merged**